ocpmodels/preprocessing/atoms_to_graphs.py

```diff
@@ -57,101 +52,44 @@ def __init__(
         self,
         max_neigh=12,
         radius=6,
-        dummy_distance=7,
-        dummy_index=-1,
         r_energy=False,
         r_forces=False,
         r_distances=False,
         r_fixed=True,
     ):
         self.max_neigh = max_neigh
         self.radius = radius
-        self.dummy_distance = dummy_distance
-        self.dummy_index = dummy_index
         self.r_energy = r_energy
         self.r_forces = r_forces
         self.r_distances = r_distances
         self.r_fixed = r_fixed
 
     def _get_neighbors_pymatgen(self, atoms):
-        """Preforms nearest neighbor search and returns split neighbors indices and distances"""
+        """Preforms nearest neighbor search and returns edge index, distances,
+        and cell offsets"""
         struct = AseAtomsAdaptor.get_structure(atoms)
-        # these return jagged arrays meaning certain atoms have more neighbors than others
-        _c_index, n_index, _images, n_distance = struct.get_neighbor_list(
-            r=self.radius
+        _c_index, _n_index, _offsets, n_distance = struct.get_neighbor_list(
+            r=self.radius, numerical_tol=0, exclude_self=True
         )
-        # find the delimiters, the number of neighbors varies
-        delim = np.where(np.diff(_c_index))[0] + 1
-        # split the neighbor index, distance, lattice offsets based on delimiter
-        split_n_index = np.split(n_index, delim)
-        split_n_distances = np.split(n_distance, delim)
-        split_offsets = np.split(_images, delim)
 
-        return split_n_index, split_n_distances, split_offsets
+        return _c_index, _n_index, n_distance, _offsets
 
-    def _pad_arrays(
-        self, atoms, split_n_index, split_n_distances, split_offsets
-    ):
-        """Pads arrays to standardize the length"""
-        # c_index is the center index, the atom to find the neighbors of
-        c_index = np.arange(0, len(atoms), 1)
-        # pad c_index
-        pad_c_index = np.repeat(c_index, self.max_neigh)
-        # add dummy variables to desired array length
-        pad_n_index = np.full((len(atoms), self.max_neigh), self.dummy_index)
-        pad_distances = np.full(
-            (len(atoms), self.max_neigh), float(self.dummy_distance)
-        )
-        pad_offsets = np.zeros((len(atoms), self.max_neigh, 3), dtype=np.int)
+    def _reshape_features(self, c_index, n_index, n_distance, offsets):
+        """Stack center and neighbor index and reshapes distances,
+        takes in np.arrays and returns torch tensors"""
+        edge_index = torch.LongTensor(np.vstack((c_index, n_index)))
+        edge_distances = torch.FloatTensor(n_distance)
+        cell_offsets = torch.LongTensor(offsets)
```

Comment:
Are we no longer setting a max no. of neighbors (`self.max_neigh`)?

Reply:
Good catch, meant to include that. 

How do we feel about leaving it unconstrained? Plans for the preprocessing is to leave max_neighbors at 200 instead of 12 (more consistent with the experiments without pbc) so this is never really hit. But I can see it being good to have.

Reply:
Yeah, having the logic in there with defaults set to 200 sounds good to me.